securedrop/journalist.py

```diff
@@ -188,6 +184,9 @@ def admin_add_user():
                 flash('There was an error with the autogenerated password. '
                       'User not created. Please try again.', 'error')
                 form_valid = False
+            except InvalidUsernameException as e:
+                form_valid = False
+                flash('Invalid username: ' + str(e), "error")
```

Comment:
`str(e)` will include the message saying that the username is too short and how long it needs to be. The username could be invalid for other reasons in the future (white space, uncode, who knows), so this is future proof.

You are right about the `admin_edit_user` block.